### PR TITLE
fix: remove utilsmodule, hashservice, refactor unique id approach

### LINKED ISSUE #836: Unique id audit
#### Is this a bug, enhancement, or feature request?
Currently our ids are not guaranteed to be unique.

### PR DESCRIPTION
**Pull request by mikerodonnell89**
fixes #836

### CODE REVIEW

File: library/src/lib/alert/alert.component.ts [hunk 1]
```diff
@@ -225,3 +223,5 @@ export class AlertComponent extends AbstractFdNgxClass implements OnInit, AfterV
     }
 
 }
+
+let id = 0;
```
Comment (MattL75) [inline library/src/lib/alert/alert.component.ts]: put these at the top, they will be easier to see and understand

File: library/src/lib/alert/alert.component.ts [hunk 2]
```diff
@@ -115,7 +112,8 @@ export class AlertComponent extends AbstractFdNgxClass implements OnInit, AfterV
     /** @hidden */
     ngOnInit(): void {
         if (!this.id) {
-            this.id = this.hasher.hash();
+            this.id = 'fd-alert-' + id;
+            id++;
```
Comment (MattL75) [inline library/src/lib/alert/alert.component.ts]: We can remove all this now and just have the id input be:

```
@Input()
id: string = 'fd-alert-' + id++;
```

File: library/src/lib/calendar/calendar.component.ts [hunk 3]
```diff
@@ -952,7 +951,8 @@ export class CalendarComponent implements OnInit, OnDestroy, AfterViewChecked, C
 
         this.constructCalendar();
         this.constructCalendarYearsList();
-        this.calendarId = this.hasher.hash();
+        this.calendarId = 'fd-calendar-' + calendarId;
+        calendarId++;
```
Comment (MattL75) [inline library/src/lib/calendar/calendar.component.ts]: Same comment as alert

File: library/src/lib/calendar/calendar.component.ts [hunk 4]
```diff
@@ -1082,3 +1081,5 @@ export class CalendarComponent implements OnInit, OnDestroy, AfterViewChecked, C
         // Will also need to subscribe to labelsChange when we go to OnPush change detection.
     }
 }
+
+let calendarId = 0;
```
Comment (MattL75) [inline library/src/lib/calendar/calendar.component.ts]: same comment as alert

File: library/src/lib/list/list-checkbox.component.ts [hunk 5]
```diff
@@ -56,14 +55,11 @@ export class ListCheckboxComponent implements OnInit, ControlValueAccessor {
...
-    /** @hidden */
-    constructor(private hash: HashService) {}
-
     /** @hidden */
     ngOnInit(): void {
         if (!this.id) {
-            this.id = this.hash.hash();
-
+            this.id = 'fd-list-checkbox-' + id;
+            id++;
```
Comment (MattL75) [inline library/src/lib/list/list-checkbox.component.ts]: same here

File: library/src/lib/list/list-checkbox.component.ts [hunk 6]
```diff
@@ -100,3 +96,5 @@ export class ListCheckboxComponent implements OnInit, ControlValueAccessor {
     }
 
 }
+
+let id = 0;
```
Comment (MattL75) [inline library/src/lib/list/list-checkbox.component.ts]: same as alert

File: library/src/lib/popover/popover.component.ts [hunk 7]
```diff
@@ -114,12 +113,12 @@ export class PopoverComponent implements OnInit {
...
 
-    /** @hidden */
-    constructor(private hasher: HashService) {}
-
     /** @hidden */
     ngOnInit(): void {
-        this.id = this.hasher.hash();
+        if (!this.id) {
+            this.id = 'fd-popover-' + id;
+            id++;
```
Comment (MattL75) [inline library/src/lib/popover/popover.component.ts]: same as alert

File: library/src/lib/popover/popover.component.ts [hunk 8]
```diff
@@ -161,3 +160,5 @@ export class PopoverComponent implements OnInit {
     }
 
 }
+
+let id = 0;
```
Comment (MattL75) [inline library/src/lib/popover/popover.component.ts]: same as alert

File: library/src/lib/tabs/tab/tab-panel.component.ts [hunk 9]
```diff
@@ -50,13 +48,13 @@ export class TabPanelComponent implements OnInit {
...
 
-    /** @hidden */
-    constructor(private hasher: HashService) {}
-
     /** @hidden */
     ngOnInit(): void {
         if (!this.id) {
-            this.id = this.hasher.hash();
+            this.id = 'fd-tab-panel-' + id;
+            id++;
```
Comment (MattL75) [inline library/src/lib/tabs/tab/tab-panel.component.ts]: same as alert

File: library/src/lib/tabs/tab/tab-panel.component.ts [hunk 10]
```diff
@@ -50,13 +48,13 @@ export class TabPanelComponent implements OnInit {
...
         }
     }
 }
+
+let id = 0;
```
Comment (MattL75) [inline library/src/lib/tabs/tab/tab-panel.component.ts]: same as alert

File: library/src/lib/toggle/toggle.component.ts [hunk 11]
```diff
@@ -68,13 +67,11 @@ export class ToggleComponent implements OnInit, ControlValueAccessor {
...
 
-    /** @hidden */
-    constructor(private hasher: HashService) {}
-
     /** @hidden */
     ngOnInit() {
         if (!this.id) {
-            this.id = this.hasher.hash();
+            this.id = 'fd-toggle-' + id;
+            id++;
```
Comment (MattL75) [inline library/src/lib/toggle/toggle.component.ts]: same as alert

File: library/src/lib/toggle/toggle.component.ts [hunk 12]
```diff
@@ -106,35 +103,37 @@ export class ToggleComponent implements OnInit, ControlValueAccessor {
...
-     * @hidden 
+     * @hidden
      * @param value Sets the value of the *checked* property of the toggle.
      */
     writeValue(value: any) {
         this.checked = value;
     }
 
-    /** 
-     * @hidden
```
Comment (MattL75) [inline library/src/lib/toggle/toggle.component.ts]: what are all these space removals for?

File: library/src/lib/toggle/toggle.component.ts [hunk 13]
```diff
@@ -106,35 +103,37 @@ export class ToggleComponent implements OnInit, ControlValueAccessor {
...
+     * @hidden
      * @param isDisabled Sets the value of the *disabled* property of the toggle.
      */
     setDisabledState(isDisabled: boolean): void {
         this.disabled = isDisabled;
     }
 
 }
+
+let id = 0;
```
Comment (MattL75) [inline library/src/lib/toggle/toggle.component.ts]: same as alert

**Comment by MattL75** [comment]
Also maybe rename the 'id' elements to something like alertUniqueId, just so that it's clear what the purpose is, and also won't conflict with anything if its exported for some reason

File: library/src/lib/toggle/toggle.component.ts (hunk 12 above)
Comment (mikerodonnell89) [reply library/src/lib/toggle/toggle.component.ts]: I think my IDE's linter does this automatically

File: library/src/lib/alert/alert.component.ts [hunk 14]
```diff
@@ -16,11 +16,12 @@ import {
     Output,
     EventEmitter, ViewEncapsulation, HostListener
 } from '@angular/core';
-import { HashService } from '../utils/hash.service';
 import { AlertRef } from './alert-utils/alert-ref';
 import { alertFadeNgIf } from './alert-utils/alert-animations';
 import { AbstractFdNgxClass } from '../utils/abstract-fd-ngx-class';
 
+let alertUniqueId = 0;
```
Comment (MattL75) [inline library/src/lib/alert/alert.component.ts]: add number type, even if its implicit its better to be explicit generally

**Pull request merged**